src/allocator.cpp

```diff
@@ -0,0 +1,123 @@
+/*
+This is a part of Loops project.
+Distributed under Apache 2 license.
+See https://github.com/vpisarev/loops/LICENSE
+*/
+
+#include "allocator.hpp"
+#include <string>
+#include <unistd.h>
+#include <sys/mman.h>
+#include <libkern/OSCacheControl.h>
+
+
+namespace loops
+{
+uint8_t* nonsys_alloc::allocate(size_t a_size)
+{
+    m_storage.push_back(std::vector<uint8_t>(a_size, 0));
```

Comment:
looks like single-connected list is emulated here, but such things are very, very error-prone. Once I spent several days trying to fix floating-error occured because of using pointers together with vectors. Please, don't do it. Use `vector<vector<>>` and use indices, avoid pointers in this place.